tests/data_group_test.py

```diff
@@ -252,12 +252,10 @@ def test_getitem_positional_indexing_without_dim_label_raises_unless_1d():
         dg2d[::]
 
 
-def test_getitem_positional_indexing_raises_when_length_is_not_unique():
+def test_getitem_positional_indexing_does_not_raise_when_length_is_not_unique():
     dg = sc.DataGroup({'a': sc.arange('x', 4), 'b': sc.arange('x', 5)})
-    with pytest.raises(sc.DimensionError):
-        dg['x', 2]
-    with pytest.raises(sc.DimensionError):
-        dg['x', 2:3]
+    dg['x', 2]
+    dg['x', 2:3]
```

Comment:
Can you also add a test where the range selected differs for the different entries?
E.g. `dg['x', 2:]` (as in the issue description).

As Numpy, we also allow slicing variables with a range end that exceeds the dimension length. Could you also check for that? I mean: `dg['x', 2:10]`

Reply:
Yes, please check:
- Some or all items of length 0
- Slicing from back
- Partially or fully exceeded ranges